pkg/daemon/ceph/osd/nsenter.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package osd
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+)
+
+const (
+	// mountNsPath is the default mount namespace of the host
+	mountNsPath = "/rootfs/proc/1/ns/mnt"
+	// nsenterCmd is the nsenter command
+	nsenterCmd = "nsenter"
+)
+
+var (
+	binPathsToCheck = []string{"/usr/sbin", "/sbin/"}
+)
+
+// NSEnter is an nsenter object
+type NSEnter struct {
+	context    *clusterd.Context
+	binary     string
+	binaryArgs []string
+}
+
+// NewNsenter returns an instance of the NSEnter object
+func NewNsenter(context *clusterd.Context, binary string, binaryArgs []string) *NSEnter {
+	return &NSEnter{
+		context:    context,
+		binary:     binary,
+		binaryArgs: binaryArgs,
+	}
+}
+
+func (ne *NSEnter) buildNsEnterCLI(binPath string) []string {
+	baseArgs := []string{fmt.Sprintf("--mount=%s", mountNsPath), "--", binPath}
+	baseArgs = append(baseArgs, ne.binaryArgs...)
+
+	return baseArgs
+}
+
+func (ne *NSEnter) callNsEnter(binPath string) error {
+	args := ne.buildNsEnterCLI(binPath)
+	op, err := ne.context.Executor.ExecuteCommandWithCombinedOutput(nsenterCmd, args...)
+	if err != nil {
+		return errors.Wrapf(err, "failed to execute nsenter. output: %s", op)
```

Comment:
I think adding the command and args called here could be good for debugging any issues unless there is a potential for the command to leak security information. 

At minimum, it might be good to include the `nsenter --mount=/rootfs/proc/1/ns/mnt -- <binPath>` information. Args could potentially contain secrets, but I don't think the part here would.